pkg/api/types.go

```diff
@@ -1802,7 +1826,25 @@ const MaxSecretSize = 1 * 1024 * 1024
 type SecretType string
 
 const (
-	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+	// SecretTypeOpaque is the default; arbitrary user-defined data
+	SecretTypeOpaque SecretType = "Opaque"
+
+	// SecretTypeServiceAccountToken contains a token that identifies a service account to the API
```

Comment:
I'm interested in this change because I'm trying to get #7444 done. I don't understand why you need a specific secret type for a service account token. Shouldn't the secret only contain a token, and the system will authorize accordingly? Why does the secret need to know it's got a service account token?